Tests/DistributedActorsTests/DeadLetterTests.swift

```diff
@@ -80,4 +81,82 @@ final class DeadLetterTests: ClusterSystemXCTestCase {
         try self.logCapture.awaitLogContaining(self.testKit, text: "This is a question")
         try self.logCapture.awaitLogContaining(self.testKit, text: "/user/ludwig")
     }
+
+    func test_remoteCallTerminatedTarget_shouldResultInDeadLetter() async throws {
+        let local = await setUpNode("local") { settings in
+            settings.enabled = true
+        }
+        let remote = await setUpNode("remote") { settings in
+            settings.enabled = true
+        }
+        local.cluster.join(node: remote.cluster.uniqueNode)
+
+        var greeter: Greeter? = Greeter(actorSystem: local)
+        let localGreeter = try Greeter.resolve(id: greeter!.id, using: remote)
+
+        let p = self.testKit.makeTestProbe(expecting: String.self)
+        let watcher = GreeterWatcher(probe: p, actorSystem: local)
+        try await watcher.watch(greeter!)
+
+        greeter = nil
+        try p.expectMessage(prefix: "Received terminated: /user/Greeter")
```

Comment:
Oh we're missing API to `p.watch` a DA from a TestProbe it seems? If you want that'd be a great thing to implement...

We'd want to be able to `p.watch(watcher); p.expectTerminated()`

Reply:
Sure. Created https://github.com/apple/swift-distributed-actors/issues/990